osf/models/preprint_service.py

```diff
@@ -74,11 +74,25 @@ def deep_url(self):
 
     @property
     def url(self):
-        return '/{}/'.format(self._id)
+        if self.provider._id == 'osf' or self.provider.domain:
+            return '/{}/'.format(self._id)
+
+        return '/preprints/{}/{}/'.format(self.provider._id, self._id)
+
+    if settings.DEV_MODE:
```

Comment:
@brianjgeiger @binoculars How about we add a `LOCAL_MODE` in settings. In this case, the staging and production environment can both use the `else` statement below.